base/logging.c

```diff
@@ -383,6 +383,55 @@ gvm_log_unlock (void)
   g_mutex_unlock (logger_mutex);
 }
 
+char *reference = NULL;
+
+/**
+ * @brief Set the log reference object.
+ *
+ * In order to be able to see which logs are related to each other, we define a
+ * common reference for them. E.g. when multiple scans in OpenVAS are running
+ * simultaniousely it is possible to detect all log corresponding to the same
+ * scan. The log reference is optional and must be set before calling
+ * setup_log_handlers. The data given must be freed by calling
+ * free_log_reference(). If called multiple times the old reference gets freed
+ * and the new one is set instead.
+ *
+ * @param ref
+ */
+void
+set_log_reference (char *ref)
```

Comment:
To make deployment of GOS less error prone and since we currently don't have semantic version as of yet maybe it makes sense to add a
```
#define LOG_REFERENCES_AVAILABLE
```

as well so that in openvas we can easily check via

```
#ifdef LOG_REFERENCES_AVAILABLE
set_log_reference(...);
#endif 
``` 
this would make it somewhat downwards compatible until gvm-libs went through.